animal-sniffer-enforcer-rule/src/main/java/org/codehaus/mojo/animal_sniffer/enforcer/CheckSignatureRule.java

```diff
@@ -330,6 +330,13 @@ private void apply( ClassFileVisitor v, File outputDirectory, MavenProject proje
                     continue;
                 }
 
+                if ( artifact.getFile() == null )
+                {
+                    logger.debug( "Skipping classes in artifact " + artifactId( artifact )
```

Comment:
well it's not a normal situation. how did you get that?

Reply:
I believe it was triggered via the pluginExecution configuration for eclipse

```xml
    <pluginExecution>
                                                <pluginExecutionFilter>
                                                    <groupId>org.apache.maven.plugins</groupId>
                                                    <artifactId>maven-enforcer-plugin</artifactId>
                                                    <versionRange>[1.0.0,)</versionRange>
                                                    <goals>
                                                        <goal>enforce</goal>
                                                    </goals>
                                                </pluginExecutionFilter>
                                                <action>
                                                    <execute>
                                                        <runOnIncremental>false</runOnIncremental>
                                                    </execute>
                                                </action>
                                            </pluginExecution>

```

Reply:
@olamy as it's not a normal situation how about we change the log level to warn or error? The aim here is to get information on what's wrong instead of an NPE.